vast/opensetAlgos/WPL.py

```diff
@@ -0,0 +1,172 @@
+    :param args: A named tuple or an argument parser object containing the arguments mentioned in the WPL_Params function above.
+    :param gpu: An integer corresponding to the gpu number to use by the current process.
+    :param models: Not used during training, input ignored.
+    :return: Iterator(Tuple(parameter combination identifier, Tuple(class name, its evm model)))
+    """
+    dimension = None
+    for pos_cls_name in pos_classes_to_process:
+        features = features_all_classes[pos_cls_name].clone().to(f"cuda:{gpu}")
+        if dimension == None:
+            dimension = features.shape[1]
+         else:
+            assert dimension == features.shape[1]
+
+        center = torch.mean(features, dim=0).view(1,dimension).to(f"cuda:{gpu}")
+        distances = torch.abs(features - center.repeat(features.shape[0], 1))
+        for tailsize, distance_multiplier, default_shape, default_scale  in itertools.product(
+            args.tailsize, args.distance_multiplier, args.default_shape, args.default_scale
+        ):
+              weibull_list = list()
+              for k in range(dimension):
+                  weibull_model = fit_high(distances[:,k].T, distance_multiplier, tailsize, default_shape, default_scale)
+                  weibull_list.append(weibull_model)
+
+                  yield (
+                    f"TS_{tailsize}_DM_{distance_multiplier:.4f}_SC_{default_scale:.4f}_SH_{default_shape:.4f}",
+                    (pos_cls_name,  {'center':center, 'weibull_list': weibull_list})
+                  )
+                      
+              
+              
+              
+def WPL_Inference(
+    pos_classes_to_process: List[str],
+    features_all_classes: Dict[str, torch.Tensor],
+    args,
+    gpu: int,
+    models: Dict = None,
+) -> Iterator[Tuple[str, Tuple[str, torch.Tensor]]]:
+    """
+    :param pos_classes_to_process: List of batches to be processed by this function in the current process.
+    :param features_all_classes: features of all classes, note the classes in pos_classes_to_process can be a subset of
+                                the keys for this dictionary
+    :param args: Can be a named tuple or an argument parser object containing the arguments mentioned in the WPL_Params
+                function above. Only the distance_metric argument is actually used during inferencing.
+    :param gpu: An integer corresponding to the gpu number to use by the current process.
+    :param models: The collated model created for a single hyper parameter combination.
+    :return: Iterator(Tuple(str, Tuple(batch_identifier, torch.Tensor)))
+    """         
+    for batch_to_process in pos_classes_to_process:
+        test_cls_feature = features_all_classes[batch_to_process].to(f"cuda:{gpu}")
+        assert test_cls_feature.shape[0] != 0
+        probs = []
+        for cls_no, cls_name in enumerate(models.keys())     
```

Comment:
While you expect the `models` variable to be an ordered dict if a user instead passes a dict rather than letting the user know this will simply provide them with incorrect results. This can be avoided by simply using  `enumerate(sorted(models.keys()))` instead of `enumerate(models.keys())`